workspaces/azure-devops/plugins/scaffolder-backend-module-azure-devops/src/actions/devopsPermitPipeline.ts

```diff
@@ -0,0 +1,159 @@
+            .describe(
+              'The API version to use for authorization. Defaults 7.1-preview.1',
+            )
+            .default('7.1-preview.1')
+            .optional(),
+        authorized: d =>
+          d
+            .boolean()
+            .describe('Whether the pipeline is authorized.')
+            .default(true),
+        pipelineId: d => d.string().describe('The ID of the pipeline.'),
+        resourceId: d => d.string().describe('The ID of the resource.'),
+        resourceType: d =>
+          d
+            .string()
+            .describe('The type of the resource (e.g. endpoint)')
+            .default('endpoint'),
+        token: d =>
+          d.string().describe('Token to use for Ado REST API.').optional(),
+      },
+    },
+    async handler(ctx) {
+      const {
+        host = 'dev.azure.com',
+        organization,
+        pipelineId,
+        project,
+        authorized,
+        resourceId,
+        resourceType,
+        apiVersion = '7.1-preview.1',
+      } = ctx.input;
+
+      const url = `https://${host}/${organization}`;
+      const credentialProvider =
+        DefaultAzureDevOpsCredentialsProvider.fromIntegrations(integrations);
+      const credentials = await credentialProvider.getCredentials({ url: url });
+
+      if (credentials === undefined && ctx.input.token === undefined) {
+        throw new InputError(
+          `No credentials provided ${url}, please check your integrations config`,
+        );
+      }
+      if (!organization) {
+        throw new InputError('organization is required');
+      }
+      if (!project) {
+        throw new InputError('project is required');
+      }
+
+      const authHandler =
+        ctx.input.token || credentials?.type === 'pat'
+          ? getPersonalAccessTokenHandler(ctx.input.token ?? credentials!.token)
+          : getBearerHandler(credentials!.token);
+
+      const webApi = new WebApi(url, authHandler);
+      const client = webApi.rest.client;
+
+      const authorizeOptions = {
+        pipelines: [
+          {
+            id: parseInt(pipelineId, 10),
+            authorized: authorized,
+          },
+        ],
+      };
+
+      // See the Azure DevOps documentation for more information about the REST API:
+      // https://learn.microsoft.com/en-us/rest/api/azure/devops/approvalsandchecks/pipeline-permissions/update-pipeline-permisions-for-resource?view=azure-devops-rest-7.1&tabs=HTTP#resourcepipelinepermissions
+      const requestUrl = `${project}/_apis/pipelines/pipelinepermissions/${resourceType}/${resourceId}?api-version=${apiVersion}`;
+      const response = await client.patch(
+        requestUrl,
+        JSON.stringify(authorizeOptions),
+        {
+          'Content-Type': 'application/json',
+          Accept: 'application/json',
+        },
+      );
+
+      if (
+        response.message.statusCode !== 200 &&
+        response.message.statusCode !== 204
+      ) {
+        ctx.logger.error('', {
```

Comment:
Is there a reason for these empty quotes? This seems off and ideally the message would be here. 🤔